source/agora/consensus/EnrollmentManager.d

```diff
@@ -226,6 +226,27 @@ public class EnrollmentManager
                                     &this.validator_set.findRecentEnrollment);
     }
 
+    /***************************************************************************
+
+        Add a enrollment data with the available height to the enrollment pool
+
+        Params:
+            enroll = the enrollment data to add
+            avail_height = block height for enrollment
+            finder = the delegate to find UTXOs with
+
+        Returns:
+            true if the enrollment data has been added to the enrollment pool
+
+    ***************************************************************************/
+
+    public bool addEnrollment (in Enrollment enroll, in Height avail_height,
+        scope UTXOFinder peekUTXO) @safe nothrow
+    {
+        return this.enroll_pool.add(enroll, avail_height, peekUTXO,
+                                    &this.validator_set.findRecentEnrollment);
+    }
+
```

Comment:
I've been trying to remove those trampoline functions from the EM for the past year.
The EM role is to manage a validator's enrollment, not to be an interface for the ValidatorSet.
Well, not anymore, although that was the original design, it proved inadequate.